test/jdk/java/awt/PrintJob/ImageTest/ImageTest.java

```diff
@@ -0,0 +1,164 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.imageio.ImageIO;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.PrintJob;
+import java.awt.Toolkit;
+import java.awt.image.BufferedImage;
+import java.awt.print.PrinterJob;
+import java.io.File;
+import java.io.IOException;
+
+import jtreg.SkippedException;
+
+/*
+ * @test
+ * @bug 4242308 4255603
+ * @key printer
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException
+ * @summary Tests printing of images
+ * @run main/manual ImageTest
+ */
+public final class ImageTest {
+
+    private static final class ImageFrame extends Frame {
+        final Image img;
+        PrintJob pjob;
+
+        private ImageFrame() {
+            super("ImageFrame");
+            img = getToolkit().getImage("image.gif");
+        }
+
+        public void paint(Graphics g) {
+            int width = img.getWidth(this);
+            int height = img.getHeight(this);
+            if (pjob != null) {
+                System.out.println("Size " + pjob.getPageDimension());
+                Dimension dim = pjob.getPageDimension();
+                if (width > dim.width) {
+                    width = dim.width - 30; // take care of paper margin
+                }
+                if (height > dim.height) {
+                    height = dim.height - 30;
+                }
+            }
+            g.drawImage(img, 10, 75, width, height, this);
+        }
+
+        public void setPrintJob(PrintJob pj) {
+            pjob = pj;
+        }
+
+        public boolean imageUpdate(Image img, int infoflags,
+                                   int x, int y, int w, int h) {
+            if ((infoflags & ALLBITS) != 0) {
+                repaint();
+                return false;
+            }
+            return true;
+        }
+    }
+
+    private static Frame init() {
+        ImageFrame f = new ImageFrame();
+        f.setLayout(new FlowLayout());
+        Button b = new Button("Print");
+        b.addActionListener(e -> {
+            PrintJob pj = Toolkit.getDefaultToolkit()
+                    .getPrintJob(f, "ImageTest", null);
+            if (pj != null) {
+                f.setPrintJob(pj);
+                Graphics pg = pj.getGraphics();
+                f.paint(pg);
+                pg.dispose();
+                pj.end();
+            }
+        });
+        f.add(b);
+        f.setBounds(0, 50, 700, 350);
```

Comment:
```suggestion
        f.setSize(700, 350);
```
The position is determined by `PassFailJFrame`.